tools/hidl/hidlcomp.cpp

```diff
@@ -4359,10 +4359,16 @@ void EspMessageInfo::write_esp()
 
         if (hasMapInfo())
             outf("\t\tdouble clientVer = ctx ? ctx->getClientVersion() : -1;\n");
+
+        const char * name = getName();
+        const char * nameEnding = &name[strlen(name) - 2];
```

Comment:
Unless there's a prohibition against single character names that is enforced elsewhere, this could be an invalid array index.

Reply:
Thank you. I don't think there's a prohibition, so I've pushed a new commit to avoid an invalid index.